app/src/main/java/com/my/journeyplanner/presenters/MainPresenter.kt

```diff
@@ -20,30 +24,91 @@ class MainPresenter(private val view: MainContract.View) : MainContract.Presente
     }
 
     override fun onPlanMyJourneyClicked() {
-        val apiService = JourneyPlannerApiService.createApiService()
-
-        call = apiService.getJourneyResults(
+        call = JourneyPlannerApiService.createApiService().getJourneyResults(
             view.getFromLocation().text.toString(),
             view.getToLocation().text.toString()
         )
 
-/*            TODO("Default adapter treats response code 300 Multiple Options as unsuccessful and")
-            TODO("puts response in the errorBody but is actually a valid response.")
-            TODO("Implement custom adapter to parse the valid response in the errorBody")   */
-        call!!.enqueue(object : Callback<JourneyPlannerDisambiguationResult> {
-            override fun onFailure(call: Call<JourneyPlannerDisambiguationResult>, throwable: Throwable) {
-                Log.e(TAG, "Other error: ${throwable.message}")
-                Log.e(TAG, "Other error: ${throwable.stackTrace.forEach { println(it) }}")
+        call!!.enqueue(object : Callback<JourneyPlanner.ItineraryResult> {
+            override fun onFailure(
+                call: Call<JourneyPlanner.ItineraryResult>,
+                throwable: Throwable
+            ) {
+                if (throwable is IOException) {
+                    Log.d(TAG, "*** Network failure ***")
+                } else {
+                    Log.d(TAG, "*** Conversion issue ***")
+                }
             }
 
             override fun onResponse(
```

Comment:
this method is wayyyy too big. Start breaking it down

Reply:
De-duped and functionalised

Reply:
Logic no longer required due to implementation of MultiOptionResponseCodeInterceptor() and JourneyPlannerResultConverterFactory